File: msgraph-mail/typescript/src/users/messages/attachmentsResponse.ts
import {SerializationWriter, ParseNode, Parsable} from '@microsoft/kiota-abstractions';
import {Attachment} from '../attachment';

export class AttachmentsResponse implements Parsable<AttachmentsResponse> {
    public value?: Attachment[] | undefined;
    public nextLink?: string | undefined;
    public serialize (writer: SerializationWriter) : void {
        writer.writeCollectionOfObjectValues<Attachment>("value", this.value);
        writer.writeStringValue("nextLink", this.nextLink);
    };
    public deserializeFields () : Map<string, (item: AttachmentsResponse, node: ParseNode) => void> {
        return new Map<string, (item: AttachmentsResponse, node: ParseNode) => void>([
            ["value", (o, n) => { o.value = n.getCollectionOfObjectValues<Attachment>(Attachment); }],
            ["nextLink", (o, n) => { o.nextLink = n.getStringValue(); }],
        ]);
    };
}

File: msgraph-mail/java/utilities/src/main/java/graphjavav4/utilities/users/mailFolders/messages/AttachmentsResponse.java
package graphjavav4.utilities.users.mailFolders.messages;

import java.util.Objects;
import graphjavav4.utilities.users.Attachment;
import com.microsoft.kiota.serialization.SerializationWriter;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.Parsable;
import java.util.function.BiConsumer;
import java.util.Map;
import java.util.HashMap;
import java.util.List;
public class AttachmentsResponse implements Parsable {
    @javax.annotation.Nullable
    public List<Attachment> value;
    @javax.annotation.Nullable
    public String nextLink;
    public void serialize(@javax.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        writer.writeCollectionOfObjectValues("value", value);
        writer.writeStringValue("nextLink", nextLink);
    }
    @javax.annotation.Nonnull
    public <T> Map<String, BiConsumer<T, ParseNode>> getDeserializeFields() {
        final Map<String, BiConsumer<T, ParseNode>> fields = new HashMap<>(2);
        fields.put("value", (o, n) -> { ((AttachmentsResponse)o).value = n.getCollectionOfObjectValues(Attachment.class); });
        fields.put("nextLink", (o, n) -> { ((AttachmentsResponse)o).nextLink = n.getStringValue(); });
        return fields;
    }
}
